storage/remote/write_handler.go

```diff
@@ -324,16 +324,85 @@ const (
 	rw20WrittenExemplarsHeader  = "X-Prometheus-Remote-Write-Written-Exemplars"
 )
 
-type responseStats struct {
-	samples    int
-	histograms int
-	exemplars  int
+// WriteResponseStats represents the response write statistics specified in https://github.com/prometheus/docs/pull/2486
```

Comment:
nit, just feels like these stats structs should exist in a different file than the write handler code :shrug:

Reply:
agree, will change

Reply:
The write client is using them as well?

Separate file is the only one that makes sense, otherwise they're either in the cient or write handler :man_shrugging: not a big deal to stick with this or change imo.